spring-integration-amqp/src/test/java/org/springframework/integration/amqp/outbound/RabbitStreamMessageHandlerTests.java

```diff
@@ -117,4 +128,79 @@ void sendNative() throws InterruptedException {
 		streamTemplate.close();
 	}
 
+	@Test
+	void errorChanelAsync() {
+		Environment env = Mockito.mock(Environment.class);
+		RabbitStreamTemplate streamTemplate = new RabbitStreamTemplate(env, "stream.stream");
+		RabbitStreamTemplate spyStreamTemplate = Mockito.spy(streamTemplate);
+		CompletableFuture<org.springframework.messaging.Message<?>> errorFuture = new CompletableFuture<>();
+		Mockito.doReturn(errorFuture).when(spyStreamTemplate).send(ArgumentMatchers.any(Message.class));
+
+		QueueChannel errorChannel = new QueueChannel();
+		RabbitStreamMessageHandler handler = RabbitStream.outboundStreamAdapter(spyStreamTemplate)
+				.sync(false)
+				.sendFailureChannel(errorChannel)
+				.getObject();
+		SimpleCodec codec = new SimpleCodec();
+		org.springframework.messaging.Message<Message> testMessage = MessageBuilder.withPayload(codec.messageBuilder()
+						.addData(new byte[1])
+						.build())
+				.build();
+		handler.handleMessage(testMessage);
+		StreamSendException streamException = new StreamSendException("Test Error Code", 99);
+		errorFuture.completeExceptionally(streamException);
+		ErrorMessage errorMessage = (ErrorMessage) errorChannel.receive(1000);
+		assertThat(errorMessage).isNotNull();
+		Throwable caughtException = errorMessage.getPayload();
+		assertThat(caughtException).isInstanceOf(StreamSendException.class);
+		assertThat(caughtException).isEqualTo(streamException);
```

Comment:
You can combine all of these into a single chained assert:
```
assertThat(errorMessage).extracting(Message::getPayload).isEqualTo(streamException);
```
I don't see a reason in extra `isInstanceOf()` since that `isEqualTo()` does cover this as well.